### PR TITLE
Apply ruff SIM101 rule: merge isinstance calls

### PR DESCRIPTION
**Pull request by Skylion007**
#### Brief description of what is fixed or changed
Merges isinstance calls and turns on SIM101 ruff check. This coding style is more readable and more performant.

#### Other comments
I will look into turning out flake8-SIM checks on in the future.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/function.py [hunk 1]
```diff
@@ -3231,11 +3230,7 @@ def count_ops(expr, visual=False):
                 # count the args
                 ops.append(o*(len(a.args) - 1))
             elif a.args and (
-                    a.is_Pow or
-                    a.is_Function or
-                    isinstance(a, Derivative) or
-                    isinstance(a, Integral) or
-                    isinstance(a, Sum)):
+                    a.is_Pow or a.is_Function or isinstance(a, (Derivative, Integral, Sum))):
```
Comment (asmeurer) [inline sympy/core/function.py:3233]: is_Pow is the same as isinstance(Pow) so this could all be combined into a single isinstance call.

**Comment by asmeurer** [comment]
This all looks fine to me.

File: sympy/core/function.py (hunk 1 above)
Comment (Skylion007) [reply sympy/core/function.py:3233]: Is this really true in all cases? Include user subclassing? Didn't want this PR to be a functional change so I left it as is.

Comment (oscarbenjamin) [reply sympy/core/function.py:3233]: I don't think we need to worry about user subclassing but I agree that keeping a distinction between stylistic and semantic changes in different PRs is good.

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**